src/stream/Encryption.ts

```diff
@@ -149,28 +208,30 @@ async function exportCryptoKey(key, { isPrivate = false } = {}) {
 // put all static functions into EncryptionUtilBase, with exception of create,
 // so it's clearer what the static & instance APIs look like
 class EncryptionUtilBase {
-    static validatePublicKey(publicKey) {
-        if (typeof publicKey !== 'string' || !publicKey.startsWith('-----BEGIN PUBLIC KEY-----')
-            || !publicKey.endsWith('-----END PUBLIC KEY-----\n')) {
-            throw new Error('"publicKey" must be a PKCS#8 RSA public key as a string in the PEM format')
+    static validatePublicKey(publicKey: crypto.KeyLike) {
+        const keyString = typeof publicKey === 'string' ? publicKey : publicKey.toString('utf8')
+        if (typeof keyString !== 'string' || !keyString.startsWith('-----BEGIN PUBLIC KEY-----')
+            || !keyString.endsWith('-----END PUBLIC KEY-----\n')) {
+            throw new Error('"publicKey" must be a PKCS#8 RSA public key in the PEM format')
         }
     }
 
-    static validatePrivateKey(privateKey) {
-        if (typeof privateKey !== 'string' || !privateKey.startsWith('-----BEGIN PRIVATE KEY-----')
-            || !privateKey.endsWith('-----END PRIVATE KEY-----\n')) {
-            throw new Error('"privateKey" must be a PKCS#8 RSA public key as a string in the PEM format')
+    static validatePrivateKey(privateKey: crypto.KeyLike) {
+        const keyString = typeof privateKey === 'string' ? privateKey : privateKey.toString('utf8')
+        if (typeof keyString !== 'string' || !keyString.startsWith('-----BEGIN PRIVATE KEY-----')
+            || !keyString.endsWith('-----END PRIVATE KEY-----\n')) {
+            throw new Error('"privateKey" must be a PKCS#8 RSA public key in the PEM format')
         }
     }
 
-    static validateGroupKey(groupKey) {
+    static validateGroupKey(groupKey: any) {
```

Comment:
Maybe remove this method? We could always use `GroupKey.validate()` directly.

Reply:
Agreed, this is leftover from previous implementation.

Reply:
Made this change in 9bd0bae9